source/Calamari.Common/Features/Packages/Decorators/ArchiveLimits/EnforceCompressionRatioDecorator.cs

```diff
@@ -30,11 +31,12 @@ bool TryVerifyCompressionRatio(string packageFile, out ArchiveLimitException? ex
             {
                 if (maximumCompressionRatio >= 1)
                 {
+                    var archiveInfo = new FileInfo(packageFile);
                     using (var archive = ArchiveFactory.Open(packageFile))
                     {
+                        var compressedSize = archiveInfo.Length;
                         var uncompressedSize = archive.TotalUncompressSize;
-                        var compressedSize = archive.TotalSize;
-                        var compressionRatio = uncompressedSize == 0 ? 0 : uncompressedSize / compressedSize;
+                        var compressionRatio = uncompressedSize == 0 ? 0 : (double)uncompressedSize / compressedSize;
```

Comment:
Hi Matt, it looks to me that if `uncompressedSize == 0`, then the ratio would just automatically be `0`, so I am not sure what this ternary expression is aiming for here. I do see if `compressedSize == 0`, then there would be a division by zero exception. If we wanted to guard against this exception, should it be `var compressionRatio = compressedSize == 0 ? 0 : (double)uncompressedSize / compressedSize;` ?

Maybe there is an assumption that when `compressedSize` is 0, `upcompressedSize` is 0 as well, but I think this could be more explicit.

Reply:
You're very right: the coalesce is intended to prevent divide-by-zero, but I got them the wrong way around. I'll change it :)